core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -0,0 +1,206 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.storage.IntsRef;
+import com.graphhopper.util.PMap;
+
+import java.util.*;
+
+import static com.graphhopper.routing.util.PriorityCode.*;
+
+/**
+ * A flag encoder for wheelchairs.
+ * <p>
+ *
+ * @author don-philipe
+ */
+public class WheelchairFlagEncoder extends FootFlagEncoder {
+    static final int SLOW_SPEED = 2;
+    static final int MEAN_SPEED = 5;
+    static final int FERRY_SPEED = 15;
+    final Set<String> safeHighwayTags = new HashSet<>();
+    final Set<String> allowedHighwayTags = new HashSet<>();
+    final Set<String> avoidHighwayTags = new HashSet<>();
+    final Set<String> avoidSurfaces = new HashSet<>();
+    final Set<String> avoidSmoothness = new HashSet<>();
+
+    /**
+     * Should be only instantiated via EncodingManager
+     */
+    public WheelchairFlagEncoder() {
+        this(4, 1);
+    }
+
+    public WheelchairFlagEncoder(PMap properties) {
+        this((int) properties.getLong("speedBits", 4),
+                properties.getDouble("speedFactor", 1));
+        this.properties = properties;
+        this.setBlockFords(properties.getBool("block_fords", true));
+    }
+
+    public WheelchairFlagEncoder(String propertiesStr) {
+        this(new PMap(propertiesStr));
+    }
+
+    public WheelchairFlagEncoder(int speedBits, double speedFactor) {
+        super(speedBits, speedFactor);
+        restrictions.add("wheelchair");
+
+        setBlockByDefault(false);
+        absoluteBarriers.add("handrail");
+        absoluteBarriers.add("wall");
+        absoluteBarriers.add("turnstile");
+        potentialBarriers.add("kerb");
+        potentialBarriers.add("cattle_grid");
+        potentialBarriers.add("motorcycle_barrier");
+
+        safeHighwayTags.add("footway");
+        safeHighwayTags.add("pedestrian");
+        safeHighwayTags.add("living_street");
+        safeHighwayTags.add("residential");
+        safeHighwayTags.add("service");
+        safeHighwayTags.add("platform");
+
+        avoidHighwayTags.add("trunk");
+        avoidHighwayTags.add("trunk_link");
+        avoidHighwayTags.add("primary");
+        avoidHighwayTags.add("primary_link");
+        avoidHighwayTags.add("secondary");
+        avoidHighwayTags.add("secondary_link");
+        avoidHighwayTags.add("tertiary");
+        avoidHighwayTags.add("tertiary_link");
+        avoidHighwayTags.add("steps");
+        avoidHighwayTags.add("track");
+
+        avoidSurfaces.add("cobblestone");
+        avoidSurfaces.add("gravel");
+        avoidSurfaces.add("sand");
+
+        avoidSmoothness.add("bad");
+        avoidSmoothness.add("very_bad");
+        avoidSmoothness.add("horrible");
+        avoidSmoothness.add("very_horrible");
+        avoidSmoothness.add("impassable");
```

Comment:
This all should not be just avoided but completely excluded from the road network